Tests/UnitTests/IdentityStateTests.swift

```diff
@@ -553,38 +553,46 @@ class IdentityStateTests: XCTestCase {
                           data: nil)
 
         let xdmSharedStateExpectation = XCTestExpectation(description: "XDM shared state should be updated once")
-        let dispatchConsentExpectation = XCTestExpectation(description: "Consent request event should be dispatched")
-        var consentEvent: Event?
-        let responseEventExpectation = XCTestExpectation(description: "Event should be dispatched after reseting identities")
-        var testEventTypes: Set<String> = [EventType.edgeConsent, EventType.edgeIdentity]
-        var testEventSources: Set<String> = [EventSource.resetComplete, EventSource.updateConsent]
+
+        let dispatchedEventsExpectation = XCTestExpectation(description: "Expected dispatched events")
+        dispatchedEventsExpectation.expectedFulfillmentCount = 2
+        var dispatchedEvents: [Event] = []
         state.resetIdentifiers(event: event,
                                createXDMSharedState: { _, _ in xdmSharedStateExpectation.fulfill()
                                }, eventDispatcher: { event in
-                                // With props.advertisingIdentifier defined at reset time, both edgeConsent and edgeIdentity hits are registered
-                                XCTAssertTrue(testEventTypes.contains(event.type))
-                                testEventTypes.remove(event.type)
-                                XCTAssertTrue(testEventSources.contains(event.source))
-                                testEventSources.remove(event.source)
-
-                                consentEvent = event
-                                dispatchConsentExpectation.fulfill()
-                                responseEventExpectation.fulfill()
+                                   // With props.advertisingIdentifier defined at reset time, both edgeConsent and edgeIdentity hits are registered
+                                   dispatchedEvents += [event]
+                                   dispatchedEventsExpectation.fulfill()
                                })
 
-        wait(for: [xdmSharedStateExpectation, responseEventExpectation], timeout: 2)
-        XCTAssertTrue(testEventTypes.isEmpty)
-        XCTAssertTrue(testEventSources.isEmpty)
+        wait(for: [xdmSharedStateExpectation, dispatchedEventsExpectation], timeout: 2)
+        
+        // Tuples allow for event type and source to be explicitly paired
+        // Update consent event
+        // Reset identity request
+        var eventCheck = [
+            (type: EventType.edgeConsent, source: EventSource.updateConsent),
+            (type: EventType.edgeIdentity, source: EventSource.resetComplete)
+        ]
+        
+        for event in dispatchedEvents {
+            if let foundIndex = eventCheck.firstIndex(where: { $0.type == event.type && $0.source == event.source }) {
+                eventCheck.remove(at: foundIndex)
+            }
+        }
+        
+        XCTAssertTrue(eventCheck.isEmpty)
```

Comment:
This works, but I prefer a simplified version of this for readability, which also checks the order of the events:
```
XCTAssertEqual(2, dispatchedEvents.length)
XCTAssertEqual(EventType.edgeConsent, dispatchedEvents[0].eventType)
XCTAssertEqual(EventSource.updateConsent, dispatchedEvents[0].eventSource)
XCTAssertEqual(EventType.edgeIdentity, dispatchedEvents[1].eventType)
XCTAssertEqual(EventSource.resetComplete, dispatchedEvents[1].eventSource)
```

Reply:
Thanks for this suggestion, it looks great! I've updated the test case and comments accordingly.